Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean

```diff
@@ -393,4 +393,8 @@ theorem bot_eq_one' : (⊥ : α) = 1 :=
 #align bot_eq_one' bot_eq_one'
 #align bot_eq_zero' bot_eq_zero'
 
+@[to_additive]
+instance CanonicallyLinearOrderedCommMonoid.linearOrderedCommMonoid : LinearOrderedCommMonoid α :=
```

Comment:
Can you just change the`extends` clause instead?

Reply:
I can change the definitions to
```
/-- A canonically linear-ordered additive monoid is a canonically ordered additive monoid
    whose ordering is a linear order. -/
class CanonicallyLinearOrderedAddCommMonoid (α : Type*)
  extends CanonicallyOrderedAddCommMonoid α, LinearOrderedAddCommMonoid α
#align canonically_linear_ordered_add_monoid CanonicallyLinearOrderedAddCommMonoid

/-- A canonically linear-ordered monoid is a canonically ordered monoid
    whose ordering is a linear order. -/
class CanonicallyLinearOrderedCommMonoid (α : Type*)
  extends CanonicallyOrderedCommMonoid α, LinearOrderedCommMonoid α
#align canonically_linear_ordered_monoid CanonicallyLinearOrderedCommMonoid
```
but then none of the `@[to_additive]` theorems works.
```
application type mismatch
  LinearOrderedAddCommMonoid.toAddCommMonoid
argument has type
  LinearOrderedCommMonoid α
but function has type
  [self : LinearOrderedAddCommMonoid α] → AddCommMonoid α
```

Reply:
I think it's worth trying to diagnose why that happens

Reply:
Do you know anyone who understands how `@[to_additive]` precisely works?

Reply:
I think you just need to add `attribute [to_additive existing] LinearOrderedCommMonoid.toCommMonoid` or similar; there are other examples of this elsewhere.